NitroxModel/DataStructures/GameLogic/FireData.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using NitroxModel.DataStructures.Util;
+using ProtoBuf;
+
+namespace NitroxModel.DataStructures.GameLogic
+{
+    [Serializable]
+    [ProtoContract]
+    public class FireData
+    {
+        [ProtoMember(1)]
+        public string FireGuid { get; set; }
+
+        [ProtoMember(2)]
+        public string SerializableCyclopsGuid
+        {
+            get { return (CyclopsGuid.IsPresent()) ? CyclopsGuid.Get() : null; }
+            set { CyclopsGuid = Optional<string>.OfNullable(value); }
+        }
+        [ProtoIgnore]
+        public Optional<string> CyclopsGuid { get; set; }
+
+
+        [ProtoMember(3)]
+        public CyclopsRooms Room { get; set; }
+        /*public CyclopsRooms? SerializableRoom
```

Comment:
Lots of dead code in here.

Reply:
Legacy from testing nullable and non-nullable data. Adding a CyclopsFireData that extends FireData would resolve the issues I've had with passing Protobuf nulls.